config/initializers/friendly_id.rb

```diff
@@ -0,0 +1,92 @@
+# FriendlyId Global Configuration
+#
+# Use this to set up shared configuration options for your entire application.
+# Any of the configuration options shown here can also be applied to single
+# models by passing arguments to the `friendly_id` class method or defining
+# methods in your model.
+#
+# To learn more, check out the guide:
+#
+# http://norman.github.io/friendly_id/file.Guide.html
+
+FriendlyId.defaults do |config|
+  # ## Reserved Words
+  #
+  # Some words could conflict with Rails's routes when used as slugs, or are
+  # undesirable to allow as slugs. Edit this list as needed for your app.
+  config.use :reserved
+
+  config.reserved_words = %w(new edit index session login logout users admin
```

Comment:
This may be more to do with rails routing conventions. According to docs,
```
resources :repositories, only: [:show, :index]
```
buys you just the `/repositories` path to get to the index action, so if you try `/repositories/index`, the router doesn't know what to do. With `get 'repositories/index'`, you're explicitly telling the router to map that path to the index action.